server/src/main/java/org/elasticsearch/search/retriever/KnnRetrieverBuilder.java

```diff
@@ -104,6 +105,8 @@ public void doExtractToSearchSourceBuilder(SearchSourceBuilder searchSourceBuild
         if (preFilterQueryBuilders != null) {
             knnSearchBuilder.addFilterQueries(preFilterQueryBuilders);
         }
-        searchSourceBuilder.knnSearch(List.of(knnSearchBuilder));
+        List<KnnSearchBuilder> knnSearchBuilders = new ArrayList<>(searchSourceBuilder.knnSearch());
```

Comment:
Thanks for catching this.

I guess it makes sense, since for `query` component we also allow it to be both a part of a standard retriever and also a main search request.